### PR TITLE
mamba list for upstream CI

### PR DESCRIPTION
**Pull request by mathause**
- [ ] Closes #xxx
 - [ ] Tests added
 - [ ] Fully documented, including `CHANGELOG.rst`

I cannot reproduce #575 locally - so let's output the list of installed packages. To see if there is any difference. I removed the line breaks from the `pip install ...` commands so they are easier to copy.

### CODE REVIEW

**Review by veni-vidi-vici-dormivi — APPROVED** [review]

File: .github/workflows/upstream-dev-ci.yaml [hunk 1]
```diff
@@ -52,7 +52,10 @@ jobs:
       - name: Install mesmer
         run: |
           python -m pip install --no-deps -e .
-
+      - name: Environment info
+        run: |
+          mamba info
+          mamba list
```
Comment (mathause) [inline .github/workflows/upstream-dev-ci.yaml:58]:
```suggestion
          micromamba info
          micromamba list
```

**Comment by mathause** [comment]
Thanks - for some reason it did not report the failure: https://github.com/MESMER-group/mesmer/actions/runs/12183532822/job/33985177056?pr=576 :man_shrugging:

**Comment by veni-vidi-vici-dormivi** [comment]
> Thanks - for some reason it did not report the failure: https://github.com/MESMER-group/mesmer/actions/runs/12183532822/job/33985177056?pr=576 🤷‍♂️

Oh wow, what a great reviewer I am 😅

File: .github/workflows/upstream-dev-ci.yaml [hunk 2]
```diff
@@ -52,7 +52,10 @@ jobs:
       - name: Install mesmer
         run: |
           python -m pip install --no-deps -e .
-
+      - name: Environment info
+        run: |
+          micromamba info
+          micromamba list
```
Comment (mathause) [inline .github/workflows/upstream-dev-ci.yaml:58]:
```suggestion
          micromamba list
          python -c "import numpy; numpy.show_config()"
```

File: .github/workflows/upstream-dev-ci.yaml [hunk 3]
```diff
@@ -20,7 +20,7 @@ jobs:
 
   upstream-dev:
     name: upstream-dev
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-24.04
```
Comment (mathause) [inline .github/workflows/upstream-dev-ci.yaml:23]:
```suggestion
    runs-on: ubuntu-latest
```

**Pull request merged**